### DISCUSSION TITLE
Why SwiftLint is not installed through Homebrew?

### DISCUSSION THREAD

**Discussion by AllDmeat** [actions/virtual-environments#7430] [General questions] [answered]
Our workflow installs bunch of dependencies through homebrew, swiftlint amongst them.
On our self-hosted runners everything works as expected, while GitHub-hosted runners with `macos-12` image fail to install swiftlint.

Basically we have a Brewfile and just invoke `brew bundle` from within our job, which fails. It can't properly install swiftlint if:
1. One is already present in the system, and
2. It was installed not though homebrew.

We've made a workaround for that by moving swiftlint out of Brewfile and introducing a separate step for it's installation like this:
`which swiftlint || brew install swiftlint`

But looks like it can be simplified if macOS images would have swiftlint preinstalled via brew.

**Comment by vpolikarpov-akvelon** [comment, marked as answer]
Hey @AllDmeat. Sorry for the late response. There are some historical reasons to install swiftlint that way. I investigated it and came to a conclusion that now we can switch to homebrew. Here is PR for that, and it's already merged: https://github.com/actions/runner-images/pull/7912. Now one should only wait for the change to be released.

**Comment by kgantchev** [comment]
> Our workflow installs bunch of dependencies through homebrew, swiftlint amongst them.  

Another way to deal with this is with [FlyCI](https://www.flyci.net/?utm_source=github&utm_medium=community&utm_campaign=recommend). Recently [we have updated our images](https://www.flyci.net/blog/flyci-image-update-xcode15-swiftlint?utm_source=github&utm_medium=community&utm_campaign=recommend) to support SwiftLint out of the box so you don't have to install it. 

What other dependencies would you like to see on your macOS images?
